### PR TITLE
Add doc comment explaining difference between ProtectedAndInternal vs ProtectedOrInternal

### PR DESCRIPTION
**Pull request by YairHalberstadt**

### CODE REVIEW

File: src/Compilers/Core/Portable/Symbols/Accessibility.cs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-﻿// Licensed to the .NET Foundation under one or more agreements.
+// Licensed to the .NET Foundation under one or more agreements.
```
Comment (Youssef1313) [inline src/Compilers/Core/Portable/Symbols/Accessibility.cs:1]: Can you revert the removal of BOM character?

**Review by jaredpar — APPROVED** [review]
approved modulo BOM change

**Comment by YairHalberstadt** [comment]
oops, pushed a ton of changes from another branch accidentally 😨 

I know this is going to annoy you @jaredpar but simplest is going to be for me to just open a new PR and close this one :-)

**Comment by YairHalberstadt** [comment]
Opened https://github.com/dotnet/roslyn/pull/50228

**Pull request closed without merging**